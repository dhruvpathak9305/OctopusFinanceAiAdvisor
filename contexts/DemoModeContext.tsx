import React, { createContext, useContext, useState, useEffect } from 'react';
import { Platform } from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';

type DemoModeContextType = {
  isDemo: boolean;
  setIsDemo: (value: boolean) => void;
  toggleDemoMode: () => void;
};

const DemoModeContext = createContext<DemoModeContextType | undefined>(undefined);

// Helper function to get storage based on platform
const getStorage = () => {
  if (Platform.OS === 'web') {
    return {
      getItem: (key: string) => {
        try {
          return Promise.resolve(localStorage.getItem(key));
        } catch (error) {
          console.error('Error accessing localStorage:', error);
          return Promise.resolve(null);
        }
      },
      setItem: (key: string, value: string) => {
        try {
          localStorage.setItem(key, value);
          return Promise.resolve();
        } catch (error) {
          console.error('Error setting localStorage:', error);
          return Promise.resolve();
        }
      }
    };
  } else {
    return AsyncStorage;
  }
};

export const DemoModeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [isDemo, setIsDemo] = useState<boolean>(false); // Default to false to use real data from accounts_real table
  const storage = getStorage();

  // Toggle the demo mode
  const toggleDemoMode = () => {
    setIsDemo(prevState => !prevState);
  };

  // Save demo mode state to storage
  useEffect(() => {
    const saveDemoMode = async () => {
      try {
        await storage.setItem('octopus-finance-demo-mode', JSON.stringify(isDemo));
      } catch (error) {
        console.error('Error saving demo mode:', error);
      }
    };
    saveDemoMode();
  }, [isDemo, storage]);

  // Load demo mode state from storage on init
  useEffect(() => {
    const loadDemoMode = async () => {
      try {
        const savedDemoMode = await storage.getItem('octopus-finance-demo-mode');
        if (savedDemoMode !== null) {
          setIsDemo(JSON.parse(savedDemoMode));
        }
      } catch (error) {
        console.error('Error loading demo mode:', error);
        // Keep default value (false) if there's an error
      }
    };
    loadDemoMode();
  }, [storage]);

  return (
    <DemoModeContext.Provider value={{ isDemo, setIsDemo, toggleDemoMode }}>
      {children}
    </DemoModeContext.Provider>
  );
};

export const useDemoMode = (): DemoModeContextType => {
  const context = useContext(DemoModeContext);
  if (context === undefined) {
    throw new Error('useDemoMode must be used within a DemoModeProvider');
  }
  return context;
}; 