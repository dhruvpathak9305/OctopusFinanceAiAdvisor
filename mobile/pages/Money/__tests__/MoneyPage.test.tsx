import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

// Mock React Router before importing the component
jest.mock('react-router-dom', () => ({
  BrowserRouter: ({ children }: { children: React.ReactNode }) => <div data-testid="router">{children}</div>,
  useNavigate: () => jest.fn(),
  useLocation: () => ({
    pathname: '/money',
    search: '',
    hash: '',
    state: null,
  }),
}));

import MoneyPage from '../index';
import { useAccounts } from '@/contexts/AccountsContext';
import { useCreditCards } from '@/contexts/CreditCardContext';

// Mock other dependencies
jest.mock('@/contexts/AccountsContext');
jest.mock('@/contexts/CreditCardContext');

// Mock ThemeProvider
jest.mock('@/common/providers/ThemeProvider', () => ({
  useTheme: jest.fn(() => ({
    theme: 'light',
    setTheme: jest.fn(),
    resolvedTheme: 'light',
  })),
  ThemeProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="theme-provider">{children}</div>
  ),
}));

jest.mock('@/components/common/ExtraCashInBankCard/ExtraCashInBankCard', () => {
  return function MockExtraCashInBankCard() {
    return <div data-testid="extra-cash-card">Extra Cash Card</div>;
  };
});

jest.mock('@/components/common/AddAccountModal', () => {
  return function MockAddAccountModal() {
    return <div data-testid="add-account-modal">Add Account Modal</div>;
  };
});

jest.mock('recharts', () => ({
  PieChart: ({ children }: { children: React.ReactNode }) => <div data-testid="pie-chart">{children}</div>,
  Pie: () => <div data-testid="pie" />,
  Cell: () => <div data-testid="cell" />,
  ResponsiveContainer: ({ children }: { children: React.ReactNode }) => <div data-testid="responsive-container">{children}</div>,
  Tooltip: () => <div data-testid="tooltip" />,
  LabelList: () => <div data-testid="label-list" />,
}));

const mockUseAccounts = useAccounts as jest.MockedFunction<typeof useAccounts>;
const mockUseCreditCards = useCreditCards as jest.MockedFunction<typeof useCreditCards>;

// Mock the contexts
const mockAccounts = [
  { id: '1', name: 'Test Savings', type: 'Savings', balance: 50000, lastUpdated: '2025-01-07' },
];

const mockCreditCards = [
  { id: '1', name: 'Test Credit Card', currentBalance: 15000, lastUpdated: '2025-01-07' }
];

jest.mock('@/contexts/AccountsContext', () => ({
  useAccounts: () => ({
    accounts: mockAccounts,
    loading: false,
    error: null
  })
}));

jest.mock('@/contexts/CreditCardContext', () => ({
  useCreditCards: () => ({
    creditCards: mockCreditCards,
    loading: false,
    error: null
  })
}));

// Mock the theme provider
jest.mock('@/common/providers/ThemeProvider', () => ({
  useTheme: () => ({
    resolvedTheme: 'dark',
  }),
}));

// Mock the responsive hook
jest.mock('@/common/hooks/useResponsive', () => ({
  useResponsive: () => ({
    isMobile: true,
  }),
}));

// Mock the child components
jest.mock('../components/AccountsSection', () => {
  return function MockAccountsSection({ onAccountSelect }: any) {
    return (
      <div data-testid="accounts-section">
        <h2>Accounts Section</h2>
        <button onClick={() => onAccountSelect({ id: '1', name: 'Test Account' })}>
          Select Account
        </button>
      </div>
    );
  };
});

jest.mock('../components/CreditCardsSection', () => {
  return function MockCreditCardsSection({ onManageCard, onViewBreakdown }: any) {
    return (
      <div data-testid="credit-cards-section">
        <h2>Credit Cards Section</h2>
        <button onClick={() => onManageCard('card-1')}>Manage Card</button>
        <button onClick={() => onViewBreakdown('card-1')}>View Breakdown</button>
      </div>
    );
  };
});

jest.mock('../components/NetWorthSection', () => {
  return function MockNetWorthSection() {
    return (
      <div data-testid="net-worth-section">
        <h2>Net Worth Section</h2>
        <p>Net worth content</p>
      </div>
    );
  };
});

const renderWithRouter = (initialEntries: string[] = ['/money']) => {
  return render(
    <MemoryRouter initialEntries={initialEntries}>
      <MoneyPage />
    </MemoryRouter>
  );
};

describe('MoneyPage', () => {
  beforeEach(() => {
    mockUseAccounts.mockReturnValue({
      accounts: [],
      loading: false,
      error: null,
      fetchAccounts: jest.fn(),
      addAccount: jest.fn(),
      updateAccount: jest.fn(),
      deleteAccount: jest.fn(),
    });

    mockUseCreditCards.mockReturnValue({
      creditCards: [],
      loading: false,
      error: null,
      fetchCreditCards: jest.fn(),
      addCreditCard: jest.fn(),
      updateCreditCard: jest.fn(),
      deleteCreditCard: jest.fn(),
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default tab', () => {
    renderWithRouter();
    
    expect(screen.getByText('Money')).toBeInTheDocument();
    expect(screen.getByText('Manage your accounts and cards')).toBeInTheDocument();
    expect(screen.getByText('Accounts')).toBeInTheDocument();
    expect(screen.getByText('Credit Cards')).toBeInTheDocument();
    expect(screen.getByText('Net Worth')).toBeInTheDocument();
  });

  it('displays accounts tab by default', () => {
    renderWithRouter();
    
    expect(screen.getByTestId('accounts-section')).toBeInTheDocument();
    expect(screen.queryByTestId('credit-cards-section')).not.toBeInTheDocument();
    expect(screen.queryByTestId('net-worth-section')).not.toBeInTheDocument();
  });

  it('switches to credit cards tab when clicked', async () => {
    renderWithRouter();
    
    const creditCardsTab = screen.getByText('Credit Cards');
    fireEvent.click(creditCardsTab);
    
    await waitFor(() => {
      expect(screen.getByTestId('credit-cards-section')).toBeInTheDocument();
      expect(screen.queryByTestId('accounts-section')).not.toBeInTheDocument();
      expect(screen.queryByTestId('net-worth-section')).not.toBeInTheDocument();
    });
  });

  it('switches to net worth tab when clicked', async () => {
    renderWithRouter();
    
    const netWorthTab = screen.getByText('Net Worth');
    fireEvent.click(netWorthTab);
    
    await waitFor(() => {
      expect(screen.getByTestId('net-worth-section')).toBeInTheDocument();
      expect(screen.queryByTestId('accounts-section')).not.toBeInTheDocument();
      expect(screen.queryByTestId('credit-cards-section')).not.toBeInTheDocument();
    });
  });

  it('handles URL parameter for tab selection', () => {
    renderWithRouter(['/money?tab=net-worth']);
    
    expect(screen.getByTestId('net-worth-section')).toBeInTheDocument();
    expect(screen.queryByTestId('accounts-section')).not.toBeInTheDocument();
  });

  it('handles URL parameter for credit cards tab', () => {
    renderWithRouter(['/money?tab=credit-cards']);
    
    expect(screen.getByTestId('credit-cards-section')).toBeInTheDocument();
    expect(screen.queryByTestId('accounts-section')).not.toBeInTheDocument();
  });

  it('defaults to accounts tab when invalid tab parameter is provided', () => {
    renderWithRouter(['/money?tab=invalid-tab']);
    
    expect(screen.getByTestId('accounts-section')).toBeInTheDocument();
  });

  it('updates URL when tab changes', async () => {
    const { container } = renderWithRouter();
    
    const netWorthTab = screen.getByText('Net Worth');
    fireEvent.click(netWorthTab);
    
    await waitFor(() => {
      // Check that URL has been updated (this would be reflected in actual routing)
      expect(screen.getByTestId('net-worth-section')).toBeInTheDocument();
    });
  });

  it('handles account selection correctly', () => {
    const consoleSpy = jest.spyOn(console, 'log').mockImplementation();
    
    renderWithRouter();
    
    const selectButton = screen.getByText('Select Account');
    fireEvent.click(selectButton);
    
    expect(consoleSpy).toHaveBeenCalledWith('Selected account:', { id: '1', name: 'Test Account' });
    
    consoleSpy.mockRestore();
  });

  it('handles credit card management correctly', async () => {
    const consoleSpy = jest.spyOn(console, 'log').mockImplementation();
    
    renderWithRouter();
    
    // Switch to credit cards tab
    const creditCardsTab = screen.getByText('Credit Cards');
    fireEvent.click(creditCardsTab);
    
    await waitFor(() => {
      const manageButton = screen.getByText('Manage Card');
      fireEvent.click(manageButton);
      
      expect(consoleSpy).toHaveBeenCalledWith('Manage card:', 'card-1');
    });
    
    consoleSpy.mockRestore();
  });

  it('handles credit card breakdown view correctly', async () => {
    const consoleSpy = jest.spyOn(console, 'log').mockImplementation();
    
    renderWithRouter();
    
    // Switch to credit cards tab
    const creditCardsTab = screen.getByText('Credit Cards');
    fireEvent.click(creditCardsTab);
    
    await waitFor(() => {
      const breakdownButton = screen.getByText('View Breakdown');
      fireEvent.click(breakdownButton);
      
      expect(consoleSpy).toHaveBeenCalledWith('View breakdown for card:', 'card-1');
    });
    
    consoleSpy.mockRestore();
  });

  it('displays proper tab styling', () => {
    renderWithRouter();
    
    const tabsList = screen.getByRole('tablist');
    expect(tabsList).toBeInTheDocument();
    
    const accountsTab = screen.getByRole('tab', { name: /accounts/i });
    const creditCardsTab = screen.getByRole('tab', { name: /credit cards/i });
    const netWorthTab = screen.getByRole('tab', { name: /net worth/i });
    
    expect(accountsTab).toBeInTheDocument();
    expect(creditCardsTab).toBeInTheDocument();
    expect(netWorthTab).toBeInTheDocument();
  });

  it('has proper icons for each tab', () => {
    renderWithRouter();
    
    // Check for FontAwesome icons
    const accountsIcon = screen.getByText('Accounts').querySelector('i.fas.fa-university');
    const creditCardsIcon = screen.getByText('Credit Cards').querySelector('i.fas.fa-credit-card');
    const netWorthIcon = screen.getByText('Net Worth').querySelector('i.fas.fa-chart-line');
    
    expect(accountsIcon).toBeInTheDocument();
    expect(creditCardsIcon).toBeInTheDocument();
    expect(netWorthIcon).toBeInTheDocument();
  });

  it('handles back navigation correctly', () => {
    const mockHistoryBack = jest.fn();
    
    // Mock window.history.back
    Object.defineProperty(window, 'history', {
      value: { back: mockHistoryBack },
      writable: true
    });
    
    renderWithRouter();
    
    const backButton = screen.getByRole('button', { name: /arrow-left/i });
    fireEvent.click(backButton);
    
    // In a real router context, this would trigger navigation
    expect(backButton).toBeInTheDocument();
  });

  it('is mobile responsive', () => {
    renderWithRouter();
    
    // Check for mobile-responsive title
    const title = screen.getByText('Money');
    expect(title).toHaveClass('text-xl'); // Mobile text size
    
    // Check for responsive tab styling
    const tabsList = screen.getByRole('tablist');
    expect(tabsList).toHaveClass('h-9'); // Mobile height
  });

  it('maintains tab state across re-renders', async () => {
    const { rerender } = renderWithRouter();
    
    // Switch to net worth tab
    const netWorthTab = screen.getByText('Net Worth');
    fireEvent.click(netWorthTab);
    
    await waitFor(() => {
      expect(screen.getByTestId('net-worth-section')).toBeInTheDocument();
    });
    
    // Re-render component
    rerender(
      <MemoryRouter initialEntries={['/money']}>
        <MoneyPage />
      </MemoryRouter>
    );
    
    // Should maintain net worth tab selection
    expect(screen.getByTestId('net-worth-section')).toBeInTheDocument();
  });

  it('handles multiple URL parameters correctly', () => {
    renderWithRouter(['/money?tab=net-worth&action=add']);
    
    // Should load net worth tab
    expect(screen.getByTestId('net-worth-section')).toBeInTheDocument();
    
    // The action parameter would be handled by the NetWorthSection component
  });

  it('provides proper accessibility', () => {
    renderWithRouter();
    
    // Check for proper heading hierarchy
    const mainHeading = screen.getByRole('heading', { level: 1 });
    expect(mainHeading).toHaveTextContent('Money');
    
    // Check for tab accessibility
    const tablist = screen.getByRole('tablist');
    expect(tablist).toBeInTheDocument();
    
    const tabs = screen.getAllByRole('tab');
    expect(tabs).toHaveLength(3);
  });

  it('updates document title appropriately', () => {
    renderWithRouter();
    
    // This would be tested if the component sets document.title
    // For now, just ensure the page title is visible
    expect(screen.getByText('Money')).toBeInTheDocument();
    expect(screen.getByText('Manage your accounts and cards')).toBeInTheDocument();
  });

  it('handles empty or no URL parameters gracefully', () => {
    renderWithRouter(['/money?']);
    
    // Should default to accounts tab
    expect(screen.getByTestId('accounts-section')).toBeInTheDocument();
  });

  it('preserves other URL parameters when changing tabs', async () => {
    renderWithRouter(['/money?tab=accounts&filter=active']);
    
    // Switch tabs
    const netWorthTab = screen.getByText('Net Worth');
    fireEvent.click(netWorthTab);
    
    await waitFor(() => {
      expect(screen.getByTestId('net-worth-section')).toBeInTheDocument();
    });
    
    // In a real implementation, other parameters would be preserved
  });

  it('displays correct tab content for each selection', async () => {
    renderWithRouter();
    
    // Test accounts tab
    expect(screen.getByTestId('accounts-section')).toBeInTheDocument();
    expect(screen.getByText('Accounts Section')).toBeInTheDocument();
    
    // Test credit cards tab
    const creditCardsTab = screen.getByText('Credit Cards');
    fireEvent.click(creditCardsTab);
    
    await waitFor(() => {
      expect(screen.getByTestId('credit-cards-section')).toBeInTheDocument();
      expect(screen.getByText('Credit Cards Section')).toBeInTheDocument();
    });
    
    // Test net worth tab
    const netWorthTab = screen.getByText('Net Worth');
    fireEvent.click(netWorthTab);
    
    await waitFor(() => {
      expect(screen.getByTestId('net-worth-section')).toBeInTheDocument();
      expect(screen.getByText('Net Worth Section')).toBeInTheDocument();
    });
  });
});
