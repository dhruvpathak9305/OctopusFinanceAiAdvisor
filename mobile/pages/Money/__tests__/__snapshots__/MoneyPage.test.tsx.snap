// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyPage Snapshot Tests matches snapshot for desktop light theme 1`] = `
<div
  class="flex flex-col min-h-screen bg-background"
>
  <div
    class="sticky top-0 z-10 bg-background border-b border-border"
  >
    <div
      class="flex items-center justify-between p-4"
    >
      <div
        class="flex items-center space-x-3"
      >
        <button
          class="p-2 rounded-lg hover:bg-muted transition-colors"
        >
          <i
            class="fas fa-arrow-left text-foreground"
          />
        </button>
        <div>
          <h1
            class="font-bold text-foreground text-2xl"
          >
            Money
          </h1>
          <p
            class="text-sm text-muted-foreground"
          >
            Manage your accounts and cards
          </p>
        </div>
      </div>
    </div>
    <div
      class="px-4 pb-4"
    >
      <div
        class="w-full"
        data-orientation="horizontal"
        dir="ltr"
      >
        <div
          aria-orientation="horizontal"
          class="items-center justify-center rounded-md text-muted-foreground grid w-full grid-cols-3 h-12 bg-muted/50 p-1"
          data-orientation="horizontal"
          role="tablist"
          style="outline: none;"
          tabindex="0"
        >
          <button
            aria-controls="radix-:r3e:-content-accounts"
            aria-selected="false"
            class="inline-flex items-center justify-center whitespace-nowrap px-3 py-1.5 ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-sm font-medium rounded-lg data-[state=active]:bg-background data-[state=active]:text-primary data-[state=active]:shadow-sm"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="inactive"
            id="radix-:r3e:-trigger-accounts"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <i
              class="fas fa-university mr-2"
            />
            Accounts
          </button>
          <button
            aria-controls="radix-:r3e:-content-credit-cards"
            aria-selected="true"
            class="inline-flex items-center justify-center whitespace-nowrap px-3 py-1.5 ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-sm font-medium rounded-lg data-[state=active]:bg-background data-[state=active]:text-primary data-[state=active]:shadow-sm"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="active"
            id="radix-:r3e:-trigger-credit-cards"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <i
              class="fas fa-credit-card mr-2"
            />
            Credit Cards
          </button>
          <button
            aria-controls="radix-:r3e:-content-net-worth"
            aria-selected="false"
            class="inline-flex items-center justify-center whitespace-nowrap px-3 py-1.5 ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-sm font-medium rounded-lg data-[state=active]:bg-background data-[state=active]:text-primary data-[state=active]:shadow-sm"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="inactive"
            id="radix-:r3e:-trigger-net-worth"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <i
              class="fas fa-chart-line mr-2"
            />
            Net Worth
          </button>
        </div>
      </div>
    </div>
    <div
      class="px-4 pb-4"
    >
      <div
        class="flex items-center space-x-2 overflow-x-auto pb-2"
      >
        <span
          class="text-muted-foreground text-sm whitespace-nowrap"
        >
          Filter by:
        </span>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors text-sm bg-primary text-primary-foreground"
        >
          <i
            class="fas fa-list mr-1"
          />
          All
        </button>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors text-sm bg-muted text-muted-foreground hover:bg-muted/80"
        >
          HDFC Bank
        </button>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors text-sm bg-muted text-muted-foreground hover:bg-muted/80"
        >
          ICICI Bank
        </button>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors text-sm bg-muted text-muted-foreground hover:bg-muted/80"
        >
          State Bank of India
        </button>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors bg-green-100 text-green-700 hover:bg-green-200 dark:bg-green-900/30 dark:text-green-400 dark:hover:bg-green-900/50 text-sm"
        >
          <svg
            class="lucide lucide-plus w-3 h-3 mr-1 inline"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="M12 5v14"
            />
          </svg>
          Add Credit Card
        </button>
      </div>
    </div>
  </div>
  <main
    class="flex-1 p-4"
  >
    <div
      class="w-full"
      data-orientation="horizontal"
      dir="ltr"
    >
      <div
        aria-labelledby="radix-:r3i:-trigger-accounts"
        class="ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 mt-0"
        data-orientation="horizontal"
        data-state="inactive"
        hidden=""
        id="radix-:r3i:-content-accounts"
        role="tabpanel"
        tabindex="0"
      />
      <div
        aria-labelledby="radix-:r3i:-trigger-credit-cards"
        class="ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 mt-0"
        data-orientation="horizontal"
        data-state="active"
        id="radix-:r3i:-content-credit-cards"
        role="tabpanel"
        style="animation-duration: 0s;"
        tabindex="0"
      >
        <div
          class="space-y-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <div>
              <h2
                class="text-lg font-semibold text-foreground"
              >
                All cards
              </h2>
              <p
                class="text-sm text-muted-foreground"
              >
                As of last available statement
              </p>
            </div>
            <span
              class="text-sm text-muted-foreground"
            >
              3
               card
              s
            </span>
          </div>
          <div
            class="grid gap-4"
          >
            <div
              data-testid="credit-card-card-1"
            >
              <span
                data-testid="card-name"
              >
                HDFC Swiggy
              </span>
              <span
                data-testid="card-institution"
              >
                HDFC Bank
              </span>
              <button
                data-testid="manage-card-btn"
              >
                Manage card
              </button>
              <button
                data-testid="view-breakdown-btn"
              >
                View breakdown
              </button>
            </div>
            <div
              data-testid="credit-card-card-2"
            >
              <span
                data-testid="card-name"
              >
                ICICI Amazon Pay
              </span>
              <span
                data-testid="card-institution"
              >
                ICICI Bank
              </span>
              <button
                data-testid="manage-card-btn"
              >
                Manage card
              </button>
              <button
                data-testid="view-breakdown-btn"
              >
                View breakdown
              </button>
            </div>
            <div
              data-testid="credit-card-card-3"
            >
              <span
                data-testid="card-name"
              >
                SBI SimplyCLICK
              </span>
              <span
                data-testid="card-institution"
              >
                State Bank of India
              </span>
              <button
                data-testid="manage-card-btn"
              >
                Manage card
              </button>
              <button
                data-testid="view-breakdown-btn"
              >
                View breakdown
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="radix-:r3i:-trigger-net-worth"
        class="ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 mt-0"
        data-orientation="horizontal"
        data-state="inactive"
        hidden=""
        id="radix-:r3i:-content-net-worth"
        role="tabpanel"
        tabindex="0"
      />
    </div>
  </main>
</div>
`;

exports[`MoneyPage Snapshot Tests matches snapshot for mobile dark theme 1`] = `
<div
  class="flex flex-col min-h-screen bg-background"
>
  <div
    class="sticky top-0 z-10 bg-background border-b border-border"
  >
    <div
      class="flex items-center justify-between p-4"
    >
      <div
        class="flex items-center space-x-3"
      >
        <button
          class="p-2 rounded-lg hover:bg-muted transition-colors"
        >
          <i
            class="fas fa-arrow-left text-foreground"
          />
        </button>
        <div>
          <h1
            class="font-bold text-foreground text-xl"
          >
            Money
          </h1>
          <p
            class="text-sm text-muted-foreground"
          >
            Manage your accounts and cards
          </p>
        </div>
      </div>
    </div>
    <div
      class="px-4 pb-4"
    >
      <div
        class="w-full"
        data-orientation="horizontal"
        dir="ltr"
      >
        <div
          aria-orientation="horizontal"
          class="items-center justify-center rounded-md text-muted-foreground grid w-full grid-cols-3 h-12 bg-muted/50 p-1"
          data-orientation="horizontal"
          role="tablist"
          style="outline: none;"
          tabindex="0"
        >
          <button
            aria-controls="radix-:r3j:-content-accounts"
            aria-selected="false"
            class="inline-flex items-center justify-center whitespace-nowrap px-3 py-1.5 ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-xs font-medium rounded-lg data-[state=active]:bg-background data-[state=active]:text-primary data-[state=active]:shadow-sm"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="inactive"
            id="radix-:r3j:-trigger-accounts"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <i
              class="fas fa-university mr-2"
            />
            Accounts
          </button>
          <button
            aria-controls="radix-:r3j:-content-credit-cards"
            aria-selected="true"
            class="inline-flex items-center justify-center whitespace-nowrap px-3 py-1.5 ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-xs font-medium rounded-lg data-[state=active]:bg-background data-[state=active]:text-primary data-[state=active]:shadow-sm"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="active"
            id="radix-:r3j:-trigger-credit-cards"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <i
              class="fas fa-credit-card mr-2"
            />
            Credit Cards
          </button>
          <button
            aria-controls="radix-:r3j:-content-net-worth"
            aria-selected="false"
            class="inline-flex items-center justify-center whitespace-nowrap px-3 py-1.5 ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-xs font-medium rounded-lg data-[state=active]:bg-background data-[state=active]:text-primary data-[state=active]:shadow-sm"
            data-orientation="horizontal"
            data-radix-collection-item=""
            data-state="inactive"
            id="radix-:r3j:-trigger-net-worth"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <i
              class="fas fa-chart-line mr-2"
            />
            Net Worth
          </button>
        </div>
      </div>
    </div>
    <div
      class="px-4 pb-4"
    >
      <div
        class="flex items-center space-x-2 overflow-x-auto pb-2"
      >
        <span
          class="text-muted-foreground text-xs whitespace-nowrap"
        >
          Filter by:
        </span>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors text-xs bg-primary text-primary-foreground"
        >
          <i
            class="fas fa-list mr-1"
          />
          All
        </button>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors text-xs bg-muted text-muted-foreground hover:bg-muted/80"
        >
          HDFC Bank
        </button>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors text-xs bg-muted text-muted-foreground hover:bg-muted/80"
        >
          ICICI Bank
        </button>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors text-xs bg-muted text-muted-foreground hover:bg-muted/80"
        >
          State Bank of India
        </button>
        <button
          class="px-3 py-1.5 rounded-full whitespace-nowrap transition-colors bg-green-100 text-green-700 hover:bg-green-200 dark:bg-green-900/30 dark:text-green-400 dark:hover:bg-green-900/50 text-xs"
        >
          <svg
            class="lucide lucide-plus w-3 h-3 mr-1 inline"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="M12 5v14"
            />
          </svg>
          Add Credit Card
        </button>
      </div>
    </div>
  </div>
  <main
    class="flex-1 p-4"
  >
    <div
      class="w-full"
      data-orientation="horizontal"
      dir="ltr"
    >
      <div
        aria-labelledby="radix-:r3n:-trigger-accounts"
        class="ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 mt-0"
        data-orientation="horizontal"
        data-state="inactive"
        hidden=""
        id="radix-:r3n:-content-accounts"
        role="tabpanel"
        tabindex="0"
      />
      <div
        aria-labelledby="radix-:r3n:-trigger-credit-cards"
        class="ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 mt-0"
        data-orientation="horizontal"
        data-state="active"
        id="radix-:r3n:-content-credit-cards"
        role="tabpanel"
        style="animation-duration: 0s;"
        tabindex="0"
      >
        <div
          class="space-y-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <div>
              <h2
                class="text-lg font-semibold text-foreground"
              >
                All cards
              </h2>
              <p
                class="text-sm text-muted-foreground"
              >
                As of last available statement
              </p>
            </div>
            <span
              class="text-sm text-muted-foreground"
            >
              3
               card
              s
            </span>
          </div>
          <div
            class="grid gap-4"
          >
            <div
              data-testid="credit-card-card-1"
            >
              <span
                data-testid="card-name"
              >
                HDFC Swiggy
              </span>
              <span
                data-testid="card-institution"
              >
                HDFC Bank
              </span>
              <button
                data-testid="manage-card-btn"
              >
                Manage card
              </button>
              <button
                data-testid="view-breakdown-btn"
              >
                View breakdown
              </button>
            </div>
            <div
              data-testid="credit-card-card-2"
            >
              <span
                data-testid="card-name"
              >
                ICICI Amazon Pay
              </span>
              <span
                data-testid="card-institution"
              >
                ICICI Bank
              </span>
              <button
                data-testid="manage-card-btn"
              >
                Manage card
              </button>
              <button
                data-testid="view-breakdown-btn"
              >
                View breakdown
              </button>
            </div>
            <div
              data-testid="credit-card-card-3"
            >
              <span
                data-testid="card-name"
              >
                SBI SimplyCLICK
              </span>
              <span
                data-testid="card-institution"
              >
                State Bank of India
              </span>
              <button
                data-testid="manage-card-btn"
              >
                Manage card
              </button>
              <button
                data-testid="view-breakdown-btn"
              >
                View breakdown
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="radix-:r3n:-trigger-net-worth"
        class="ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 mt-0"
        data-orientation="horizontal"
        data-state="inactive"
        hidden=""
        id="radix-:r3n:-content-net-worth"
        role="tabpanel"
        tabindex="0"
      />
    </div>
  </main>
</div>
`;
